lib/api/api_auth_webauthn.go

```diff
@@ -448,7 +431,15 @@ func (s *WebauthnVolatileState) init() {
 	}
 }
 
+// Load volatile WebAuthn state with a read lock during loading.
 func (s *webauthnService) loadVolatileState() *WebauthnVolatileState {
+	s.volStateMut.RLock()
+	defer s.volStateMut.RUnlock()
+	return s.unsafeLoadVolatileState()
+}
+
+// Load volatile WebAuthn state without acquiring a read lock.
+func (s *webauthnService) unsafeLoadVolatileState() *WebauthnVolatileState {
```

Comment:
The usual naming in Syncthing's codebase seems to be `...RLocked()` if the function is expected to be called with a read lock held, `...Locked()` for a full / write lock.

And in the conditional below, if the data cannot be unmarshalled, shouldn't we rather discard it?  What other way is there to get rid of data that we will never be able to use again?

Reply:
> The usual naming in Syncthing's codebase seems to be `...RLocked()`

Thanks!

> And in the conditional below, if the data cannot be unmarshalled, shouldn't we rather discard it? What other way is there to get rid of data that we will never be able to use again?

Do you mean we should proactively delete/overwrite the stored data immediately on load failure? I don't think that's necessary, it'll happen automatically on the next write to it anyway. It would also require acquiring a write lock in this function, which looks like it would be messy to handle since this function is itself called in the write function (with a write lock already acquired).

Reply:
You're right it introduces unnecessarily complex code.  If there is a way to discard invalid data (on updating you say), then that's fine with me.  The user might get annoyed with the message a few times, but it's already a highly unlikely scenario to begin with.